capsules/src/net/ipv6/ipv6_send.rs

```diff
@@ -0,0 +1,158 @@
+use core::cell::Cell;
+use ieee802154::device::{MacDevice, TxClient};
+use kernel::ReturnCode;
+use kernel::common::take_cell::TakeCell;
+use net::ieee802154::MacAddress;
+use net::ipv6::ip_utils::IPAddr;
+use net::ipv6::ipv6::{IP6Header, IP6Packet, TransportHeader};
+use net::sixlowpan::sixlowpan_state::TxState;
+
+// TODO: These should *not* be constants, and should be set at some other
+// point during the initialization of the IP stack
+const SRC_MAC_ADDR: MacAddress = MacAddress::Short(0xf00f);
+const DST_MAC_ADDR: MacAddress = MacAddress::Short(0xf00e);
+
+pub trait IP6Client {
+    fn send_done(&self, result: ReturnCode);
+}
+
+pub trait IP6Sender<'a> {
+    fn set_client(&self, client: &'a IP6Client);
+
+    fn set_addr(&self, src_addr: IPAddr);
+
+    fn set_gateway(&self, gateway: MacAddress);
```

Comment:
Maybe `set_gateway` and `set_addr` should just be implementation specific (in our case, probably just in the constructor) rather than part of a generic trait.